forms-flow-web/src/components/Modeler/DecisionTable.js

```diff
@@ -0,0 +1,287 @@
+import React ,{ useState, useEffect } from "react";
+import { useSelector ,useDispatch } from 'react-redux';
+import {
+  CustomButton,
+  CustomSearch,
+    DownArrowIcon,
+} from "@formsflow/components";
+import LoadingOverlay from "react-loading-overlay-ts";
+import { useTranslation ,Translation } from "react-i18next";
+import SortableHeader from '../CustomComponents/SortableHeader';
+import { fetchAllProcesses } from "../../apiManager/services/processServices";
+import { Dropdown } from "react-bootstrap";
+import Pagination from "react-js-pagination";
+import { HelperServices } from "@formsflow/service";
+import { MULTITENANCY_ENABLED } from "../../constants/constants";
+import { push } from "connected-react-router";
+import { setDmnSearchText,setIsPublicDiagram } from "../../actions/processActions";
+
+
+const DecisionTable = React.memo(() => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const dmn = useSelector((state) => state.process?.dmnProcessList);
+  const [isLoading, setIsLoading] = useState(true);
+  const searchText = useSelector((state) => state.process?.dmnSearchText);
+  const [activePage, setActivePage] = useState(1);
+  const [limit, setLimit] = useState(5);
+  const tenantKey = useSelector((state) => state.tenants?.tenantId);
+  const totalCount = useSelector ((state) => state.process.totalDmnCount);
+  const [currentDmnSort, setCurrentDmnSort] = useState({
+    activeKey: "name", 
+    name: { sortOrder: "asc" },
+    id: { sortOrder: "asc" },
+    modified: { sortOrder: "asc" },
+    status: { sortOrder: "asc" },
+  });
+  const [searchDmnLoading, setSearchDmnLoading] = useState(false);
+  const redirectUrl = MULTITENANCY_ENABLED ? `/tenant/${tenantKey}/` : "/";
+  const [search, setSearch] = useState(searchText || "");
+
+  useEffect(() => {
+    setIsLoading(true);
+    setSearchDmnLoading(true);
```

Comment:
Is this the loader in search bar? do we need its loader if any change happens in table ? it should be loading only on ENTER button click @fahad-aot

Reply:
Updated.